### PR TITLE
add deploy env variable to config.yaml and adjust charm code

### PR DESCRIPTION
**Pull request by matheushent**

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -84,3 +84,8 @@ options:
     default:
     description: |
       The secret key for the Auth0 app client to which tokens will be issued.
+  deploy-env:
```
Comment (julianaklulo) [inline config.yaml:87]: Can you add it to the `license-manager-agent.yaml` example in the README, please?

**Review by heitorPB — COMMENTED** [review]
Why do we need a new configuration option? Why this can't be inferred by the system, e.g. by looking at the the backend url?

File: config.yaml (hunk 1 above)
Comment (matheushent) [reply config.yaml:87]: yes, I will

**Comment by matheushent** [comment]
> Why do we need a new configuration option? Why this can't be inferred by the system, e.g. by looking at the the backend url?

I wouldn't say it can't because it can, the question is: is the best approach? It's just too simple to add a new variable since the agent already expects it (https://github.com/omnivector-solutions/license-manager/blob/main/agent/lm_agent/config.py#L53).

**Comment by matheushent** [comment]
> > > Why do we need a new configuration option? Why this can't be inferred by the system, e.g. by looking at the the backend url?
> 
> > 
> 
> > I wouldn't say it can't because it can, the question is: is the best approach?
> 
> 
> 
> That's precisely my point. We can have the environment with an ever growing number of variables, but I don't feel that is a good approach. This increases the complexity of maintenance and deploys, making it more expensive and error-prone.
> 
> 
> 
> So, can't be inferred by the system?

I disagree it increases the complexity of maintaining. The purpose of the task is indeed to make it easier to understand the Sentry environments and the way the apps use them.

Answering the question indeed: it would work only for jobbergate cli and license manager agent I think, but I would need to add a parser in each application to infer it, and that indeed would increase the complexity.

To decrease the complexity in order to have less files created (each config is a file) we can plan another approach. As soon as we grow the apps it's expected to have more variables.

**Comment by matheushent** [comment]
> > > > Why do we need a new configuration option? Why this can't be inferred by the system, e.g. by looking at the the backend url?
> > 
> > > 
> > 
> > > I wouldn't say it can't because it can, the question is: is the best approach?
> > 
> > 
> > 
> > That's precisely my point. We can have the environment with an ever growing number of variables, but I don't feel that is a good approach. This increases the complexity of maintenance and deploys, making it more expensive and error-prone.
> > 
> > 
> > 
> > So, can't be inferred by the system?
> 
> I disagree it increases the complexity of maintaining. The purpose of the task is indeed to make it easier to understand the Sentry environments and the way the apps use them.
> 
> Answering the question indeed: it would work only for jobbergate cli and license manager agent I think, but I would need to add a parser in each application to infer it, and that indeed would increase the complexity.
> 
> To decrease the complexity in order to have less files created (each config is a file) we can plan another approach. As soon as we grow the apps it's expected to have more variables.

Also, let's say it can be inferred from the backend url (I can't any other way to identify), it wouldn't be possible to set for local purposes because there's no backend url like `https://armada-k8s.local.omnivector.solutions`

**Review by dusktreader — APPROVED** [review]

**Review by heitorPB — APPROVED** [review]

**Review by julianaklulo — APPROVED** [review]

**Pull request merged**